concept/thing/impl/AttributeImpl.java

```diff
@@ -67,6 +69,14 @@ private AttributeImpl(AttributeVertex<VALUE> vertex) {
 
     public abstract VALUE getValue();
 
+    @Override
+    public int compareTo(Attribute o) {
+        AttributeImpl<?> other = (AttributeImpl<?>) o;
+        if (getType().getValueType().comparables().contains(other.getType().getValueType())) {
+            return getType().getValueType().comparator(other.getType().getValueType()).compare(getValue(), other.getValue());
+        } else throw TypeDBException.of(ATTRIBUTES_NOT_COMPARABLE, this, other);
+    }
```

Comment:
we move the comparison by attribute value to `Attribute` concept rather than implementing it in the `Matcher`. We also define the Comparators used for particular value types exactly once: in the `Encoding`